auth/src/main/java/org/aerogear/mobile/auth/credentials/JwksManager.java

```diff
@@ -0,0 +1,155 @@
+
+    private HttpServiceModule httpModule;
+    private AuthServiceConfiguration authServiceConfiguration;
+    private SharedPreferences sharedPrefs;
+    private Logger logger;
+
+    JwksManager(Context context, HttpServiceModule httpModule, AuthServiceConfiguration authServiceConfiguration, Logger logger) {
+        this.httpModule = httpModule;
+        this.authServiceConfiguration = authServiceConfiguration;
+        this.sharedPrefs = context.getSharedPreferences(STORE_NAME, Context.MODE_PRIVATE);
+        this.logger = logger;
+    }
+
+    /**
+     * Load the cached JWKS from the private storage of the app.
+     * It will return null if there is no cached JWKS found.
+     * It will trigger a request to fetch the JWKS in the background if there is no cached key found, or {@link AuthServiceConfiguration#getMinTimeBetweenJwksRequests()} is passed since the key set is requested last time.
+     * @return the cached JWKS, or null if it doesn't exist
+     */
+    public JsonWebKeySet load(KeycloakConfiguration keyCloakConfig) {
+        boolean needFetchNow = true;
+        JsonWebKeySet jwks = null;
+        String namespace = keyCloakConfig.getRealmName();
+        String keyContentEntryName = buildEntryNameForKeyContent(namespace);
+        String keyContent = this.sharedPrefs.getString(keyContentEntryName, null);
+        if (keyContent != null) {
+            try {
+                jwks = new JsonWebKeySet(keyContent);
+                needFetchNow = false;
+            } catch (JoseException e) {
+                logger.error("failed to parse JsonWebKeySet content", e);
+            }
+        }
+        fetchJwksIfNeeded(keyCloakConfig, needFetchNow);
+        return jwks;
+    }
+
+    /**
+     * Fetch the JWKS from the server if necessary and save them locally.
+     * The request will be trigger if:
+     * 1. forceFetch is set to true, or
+     * 2. {@link AuthServiceConfiguration#getMinTimeBetweenJwksRequests()} is passed since the key set is requested last time.
+     * @param keycloakConfiguration the configuration of the keycloak server
+     * @param forceFetch if set to true, the request will be trigger immediately.
+     */
+    public void fetchJwksIfNeeded(KeycloakConfiguration keycloakConfiguration, boolean forceFetch) {
+        if (forceFetch || shouldRequestJwks(keycloakConfiguration)) {
+            String jwksUrl = keycloakConfiguration.getJwksUrl();
+            HttpRequest getRequest = httpModule.newRequest();
+            getRequest.get(jwksUrl);
+            HttpResponse response = getRequest.execute();
+            response.onComplete(() -> {
+                //this is invoked on a background thread.
+                if (response.getStatus() == 200) {
+                    String jwksContent = response.stringBody();
+                    boolean isValidKeyset = true;
+                    try {
+                        new JsonWebKeySet(jwksContent);
```

Comment:
Missing assignment?

Reply:
the original plan was just trying the validate the content and that's why it didn't need assignment. But now I have changed the API slightly and it will have the assignment statement.